pkg/apis/logging/v1/clusterlogging_types.go

```diff
@@ -210,6 +215,147 @@ type CuratorSpec struct {
+// https://docs.fluentd.org/configuration/buffer-section#buffering-parameters
+//
+// For flush parameters refer to:
+// https://docs.fluentd.org/configuration/buffer-section#flushing-parameters
+//
+// For retry parameters refer to:
+// https://docs.fluentd.org/configuration/buffer-section#retries-parameters
+type FluentdBufferSpec struct {
+	// ChunkLimitSize represents the maximum size of each chunk. Events will be
+	// written into chunks until the size of chunks become this size. For data
+	// type pattern see:
+	// https://docs.fluentd.org/configuration/config-file#supported-data-types-for-values
+	//
+	// Notice: The OpenAPI validation pattern is an ECMA262 regular expression
+	// (See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#properties)
+	//
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([kmgtKMGT]{0,1})$"
+	ChunkLimitSize string `json:"chunkLimitSize"`
+
+	// TotalLimitSize represents the threshold of node space allowed per fluentd
+	// buffer to allocate. Once this threshold is reached, all append operations
+	// will fail with error (and data will be lost). For data type pattern see:
+	// https://docs.fluentd.org/configuration/config-file#supported-data-types-for-values
+	//
+	// Notice: The OpenAPI validation pattern is an ECMA262 regular expression
+	// (See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#properties)
+	//
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([kmgtKMGT]{0,1})$"
+	TotalLimitSize string `json:"totalLimitSize"`
+
+	// OverflowAction represents the action for the fluentd buffer plugin to
+	// execute when a buffer queue is full.
+	//
+	// +kubebuilder:validation:Enum:=throw_exception;block;drop_oldest_chunk
+	OverflowAction OverflowActionType `json:"overflowAction"`
+
+	// FlushThreadCount reprents the number of threads used by the fluentd buffer
+	// plugin to flush/write chunks in parallel.
+	FlushThreadCount int32 `json:"flushThreadCount"`
+
+	// FlushMode represents the mode of the flushing thread to write chunks. The mode
+	// allows lazy (if `time` parameter set), per interval or immediate flushing.
+	//
+	// +kubebuilder:validation:Enum:=lazy;interval;immediate
+	FlushMode FlushModeType `json:"flushMode"`
+
+	// FlushInterval represents the time duration to wait between two consecutive flush
+	// operations. Takes only effect used together with `flushMode: interval`. For data
+	// type pattern see:
+	// https://docs.fluentd.org/configuration/config-file#supported-data-types-for-values
+	//
+	// Notice: The OpenAPI validation pattern is an ECMA262 regular expression
+	// (See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#properties)
+	//
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([smhd]{0,1})$"
+	FlushInterval string `json:"flushInterval"`
+
+	// FlushAtShutdown represents the switch to flush all buffers on fluentd shutdown.
+	FlushAtShutdown bool `json:"flushAtShutdown"`
```

Comment:
i don't think this is necessary to expose to customers. we should pick an opinionated value (likely true)